### ISSUE TITLE
typos pre-commit hook building from source rather than binary install

### ISSUE THREAD

**Issue by tekumara** [crate-ci/typos#682]
In _.pre-commit-hooks.yaml_ I have:

```yaml
  - repo: https://github.com/crate-ci/typos
    rev: v1.13.20
    hooks:
      - id: typos
```

But the install takes 3 mins because it is building from source on x86_64 linux (Ubuntu 22.04).

**Comment by tekumara** [comment]
Apparently [pre-commit hooks are installed](https://github.com/pre-commit/pre-commit/issues/1508#issuecomment-648874721) by cloning the repo and running  `pip install .`

So I can simulate this using `pip install -v git+https://github.com/crate-ci/typos.git@v1.13.20`. When I do, I can see it's compiling all the crates, rather than downloading and using the binaries.

**Comment by tekumara** [comment]
It looks like its using maturin to build the wheels:
```
  Running command Building wheel for typos (pyproject.toml)
  Running `maturin pep517 build-wheel -i /home/runner/work/aec/aec/typos/.venv/bin/python --compatibility off`
  📦 Including license file "/tmp/pip-req-build-1h1ots63/LICENSE-APACHE"
  📦 Including license file "/tmp/pip-req-build-1h1ots63/LICENSE-MIT"
  🔗 Found bin bindings
  📡 Using build options bindings from pyproject.toml
     Compiling proc-macro2 v1.0.51
     Compiling unicode-ident v1.0.6
```

I wonder if https://github.com/crate-ci/typos/pull/67, which switched to the maturin backend instead of setup.py, means setup.py from https://github.com/crate-ci/typos/pull/289 is now being ignored?

**Comment by tekumara** [comment]
If I used the version prior, it fetches the binaries and is a lot faster:
```
pip install -v git+https://github.com/crate-ci/typos.git@v1.13.18
....
running fetch_binaries
....
```

**Comment by tekumara** [comment]
If I use the version prior, it fetches the binaries and is a lot faster:
```
pip install -v git+https://github.com/crate-ci/typos.git@v1.13.18
....
running fetch_binaries
....
```

**Comment by CalumY** [comment]
For reference, Ruff solves this by making a pre-commit mirror: https://github.com/charliermarsh/ruff-pre-commit.

**Comment by epage** [comment]
Huh, that was an unintended side effect of adding PyPI packages.

I've been avoiding specialized repos for pre-commit and Github Actions to keep the release process automated within existing tooling.

**Comment by CalumY** [comment]
I created a mirror at https://github.com/CalumY/typos-pre-commit. 

It can be added to your pre-commit config with: 

```
  - repo: https://github.com/calumy/typos-pre-commit
    rev: 'v1.13.20'
    hooks:
      - id: typos
```

I'm happy to transfer this to @create-ci if that would be prefered.

**Comment by CalumY** [comment]
However, if #685 fixes this, there is no need for this mirror.

**Comment by epage** [comment]
Can someone confirm that that#685 fixed things?

**Comment by CalumY** [comment]
I don't think it did, I'm afraid. For pre-commit work for a python package you have to be able to install it with `pip install .`. When I ran this command I got the following error:

```
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [14 lines of output]
      error: Multiple top-level packages discovered in a flat-layout: ['docker', 'crates', 'benchsuite'].
      
      To avoid accidental inclusion of unwanted files or directories,
      setuptools will not proceed with this build.
      
      If you are trying to create a single distribution with multiple packages
      on purpose, you should not rely on automatic discovery.
      Instead, consider the following options:
      
      1. set up custom discovery (`find` directive with `include` or `exclude`)
      2. use a `src-layout`
      3. explicitly set `py_modules` or `packages` with a list of names
      
      To find more information, look for "package discovery" on setuptools docs.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

When running pre-commit with the following config it created a similar error. 

```
  - repo: https://github.com/crate-ci/typos
    rev: 5f7454815c1e60353d4fdd872c86982dc0d3a99b
    hooks:
      - id: typos
```

**Comment by CalumY** [comment]
Suggested a fix in #686.
